### ISSUE TITLE
winget (Windows Package Manager) に登録してほしい

### ISSUE THREAD

**Issue by juner** [sakura-editor/sakura#1700]
## (必須) やりたいこと(=実現したいこと)
winget で インストール・アップデート・アンインストールできるようにしてほしい。

## (省略可) 解決手段の提案
windows package manager(winget) に登録する
https://docs.microsoft.com/windows/package-manager/package/

**Comment by kazasaku** [comment]
はじめまして。

wingetというコマンドラインツールで導入・更新できるようにしてほしい、という要望でよろしいでしょうか？

なお、更新方法や新バージョンの通知方法については、それぞれ #78 ・ #1131 にて検討されています。

**Comment by juner** [comment]
@kazasaku そうです。インストーラーがあるソフトウェアだと公式リポジトリにpull request出すだけで登録できるそうなのでissueを上げました。よろしくお願いいたします。

**Comment by kazasaku** [comment]
「winget使ってみませんか」ってことですかね。なぜwingetなのか理由が説明されていないので、これ以上は認識できませんでした。

**Comment by k-takata** [comment]
今までも Chocolatey のようなサードパーティー製のパッケージマネージャーはありましたが、winget は MS の開発で、しばらく前に v1.0 も出ましたから、今後 winget が広く使われるようになることは十分あり得るでしょう。

**Comment by kazasaku** [comment]
言いたいことはいろいろあったのですが、wingetやwinget-pkgsの設計・運用方針に対するコメントになってしまうので書かないでおきます。

確かwingetはCUIツールでしたよね？
一般ユーザーのことも考えると、（個人的には）GUIなツールの方がいいと思います。
誰かがGUIフロントエンドを用意すれば別ですが。

**Comment by berryzplus** [comment]
個人的には、マイクロソフトの最新技術には飛びつくのは危険、だと思っています。

![image](https://user-images.githubusercontent.com/3253151/124834972-8a790f00-dfbb-11eb-9456-2da6e42bf8f5.png)

最低限ドキュメントの「重要」が外れるまでは手を出すべきではないと思います。
膨大な学習コストを費やしたあと「結局使えない」というケースがままあるためです。

仕様をチラ見した限り、テキトーにYAML書いたら対応できそうです。

YAMLによる定義を「マニュフェスト」って呼ぶのを最近よく見かけるようになった気がします。
生Windows APIを使う環境で「マニュフェスト」と言ったら exe に埋め込む RT_MANIFEST を指すと思います。
まぎらわしいことこの上ないです（RT_MANIFESTを知らなければ混同する危険はありませんが:smiley:

誰かがwingetの仕様を勉強して、「これで対応できそう」をPRする分には問題ないと思います。

**Comment by k-kagari** [comment]
YAML 書いて PR 出すだけなら担当してみようかと思い調べてみましたが、現在リリースされているバージョンの winget は [zip ファイルからのインストールに対応していない](https://github.com/microsoft/winget-cli/issues/140)らしく、[本プロジェクトが今のところ配布しているパッケージ](https://github.com/sakura-editor/sakura/releases)をそのまま登録することはできないようです。zip がサポートされるまで待つか、インストーラーの exe を直接配布するようにプロジェクトのリリース体制を変更する必要があります。前者は該当 issue を読む限りいつになるか分からない雰囲気ですから、本気でやるなら後者ですね。

もし目的が「パッケージマネージャーを使ってインストールすること」なら @KageShiron さん提供の Chocolatey パッケージが利用できます。 (#55)

**Comment by berryzplus** [comment]
> 最低限ドキュメントの「重要」が外れるまでは手を出すべきではないと思います。

公式ドキュメントの「重要」が外されたことを確認しました。
※重要には「wingetはpublic previewなので変更される可能性があります」と書かれていました。

#### ■手順抜粋
1. [アプリケーションに関する情報を提供するパッケージ マニフェストを作成します](https://docs.microsoft.com/ja-jp/windows/package-manager/package/manifest)。 マニフェストは、Windows パッケージ マネージャー スキーマに従う YAML ファイルです。
2. [マニフェストを Windows パッケージ マネージャー リポジトリに送信します](https://docs.microsoft.com/ja-jp/windows/package-manager/package/repository)。 これは、winget ツールがアクセスできるマニフェストのコレクションを含む GitHub のオープン ソース リポジトリです。

なんとなく、1と2は作業分担になりそうに思いました。

1を @k-kagari さんに作っていただくとして、
2を GitHub Actions で実現するのを @kazasaku さんにお願いするのかな？と思いました（他力本願）
意見、要望、文句、その他「オレがやる！」等あれば、ここに書いていく感じでお願いします。

**Comment by sheepwng** [comment]
いつも更新いただきありがとうございます。
便利に使わせていただいております。

最近Wingetを使い始めていろいろなアプリがダウンロードできてPCの環境構築に便利なのですが、本スレッドが更新されていないところを見ると対応は難しそうでしょうか。

MS製の「wingetcreate.exe」というCliツールでもマニフェストファイルを作成出来そうということはわかったのですが素人では難しかったのでご検討いいただけると幸いです。

**Comment by BHCrusher1** [comment]
自分の方でwinget用のマニフェストを作成してみました。
インストールができることは確認済みです。
https://github.com/BHCrusher1/winget-pkgs/commit/833afaec011d7419a76faf70a4b25e4ddd16962d

ここでレビューしていただけたら本家 https://github.com/microsoft/winget-pkgs の方にプルリク投げたいと考えています。
特に確認が必要な点としては、以下の点と考えています。
- PackageIdentifier
- Publisher
- PackageName
- URL系

何か議論するにあたり、このIssues以外に適切な場所があるようであれば、教えていただければ幸いです。

**Comment by berryzplus** [comment]
問題ないと思います。ありがとうございます。

強いて言うと発行者URLとパッケージURLが逆かも知れません。
- 発行者URL https://sakura-editor.github.io/
- パッケージURL https://github.com/sakura-editor/sakura

**Comment by BHCrusher1** [comment]
指摘のあった点の修正と、FileExtensionsの追加を行いました。

[PublisherUrlとPackageUrlを逆にした](https://github.com/BHCrusher1/winget-pkgs/commit/c243996b0435b689dd9ae1f7ea9b387f2dd2b903)
[インストーラーの記述を元にFileExtensionsの内容を追加](https://github.com/BHCrusher1/winget-pkgs/commit/5ca263c0aa62300e82d4449d2e1ac0566b7d1824)

問題なければwinget-pkgsにプルリク投げます。

**Comment by BHCrusher1** [comment]
wingetに登録されたみたいです。Issuesクローズできる方はお願いします。

<img width="1258" height="586" alt="Image" src="https://github.com/user-attachments/assets/a251dfb3-487f-4bc1-8e40-0e74a5ab20d9" />

**Comment by juner** [comment]
対応ありがとうございます！
